libs/server/src/lib/utils/read-projects.ts

```diff
@@ -74,14 +75,14 @@ export async function readBuilderSchema(
     path.join(npmPackage, 'package.json'),
     path.join(basedir, 'node_modules')
   );
-  const b = packageJson.json.builders;
+  const b = toLegacyFormat(packageJson.json).builders;
   const buildersPath = b.startsWith('.') ? b : `./${b}`;
   const buildersJson = readAndCacheJsonFile(
     buildersPath,
     path.dirname(packageJson.path)
   );
 
-  const builderDef = buildersJson.json.builders[builderName];
+  const builderDef = toLegacyFormat(buildersJson.json).builders[builderName];
```

Comment:
See the comment above. toLegacyFormat only updates workspace.json/angular.json. You need to a different toLegacyFormat here.